packages/client/src/forms/utils.ts

```diff
@@ -557,11 +561,33 @@ export const getConditionalActionsForField = (
   return (
     field.conditionals
       // eslint-disable-next-line no-eval
-      .filter((conditional) => eval(conditional.expression))
+      .filter((conditional) =>
+        evalExpressionInFieldDefinition(
+          conditional.expression,
+          values,
+          offlineCountryConfig,
+          draftData,
+          userDetails
+        )
+      )
       .map((conditional: Conditional) => conditional.action)
   )
 }
 
+export const evalExpressionInFieldDefinition = (
+  expression: string,
+  /*
+   * These are used in the eval expression
+   */
+  values: IFormSectionData,
+  offlineCountryConfig?: IOfflineData,
+  draftData?: IFormData,
+  userDetails?: UserDetails | null
+) => {
+  // eslint-disable-next-line no-eval
+  return eval(expression)
+}
```

Comment:
Could we name these with a prefix here

```suggestion
  $values: IFormSectionData,
  $config?: IOfflineData,
  $draft?: IFormData,
  $user?: UserDetails | null
) => {
 // For backwards compatibility
  const values = $values
  const offlineCountryConfig = $config
  const draftData = $draft
  const userDetails = $userDetails 
  // eslint-disable-next-line no-eval
  return eval(expression)
}
```

Reply:
Also, is it possible to to remove the optionality of the parameters with little effort? Not sure if there is a real case where these would not get supplied

```ts
$form: IFormSectionData
$config: IOfflineData,
$draft: IFormData,
$user: UserDetails
```